app/models/item.rb

```diff
@@ -26,4 +26,10 @@ def self.ready_to_ship
     .group("items.id, invoices.id")
     .order("invoice_creation asc")
   end
+
+  def self.merchant_invoices
+    joins(:invoices)
+    .select("invoices.id")
+    .distinct
+  end
```

Comment:
Wait -- isn't this supposed to just be invoices for the merchant?? It doesn't look like we're limiting it to that here. Are you just making that happen through the view then by calling Merchant.items and only getting invoices for those items?

Reply:
Yea, that's correct. To remove the law of demeter violation plus making sure it's only for that merchant I could just do it in the query & remove the merchant.items from the controller most likely